src/robot/libraries/BuiltIn.py

```diff
@@ -613,10 +613,21 @@ def should_be_equal(self, first, second, msg=None, values=True):
         for example, string ``false`` or ``no values``. See `Boolean arguments`
         section for more details.
 
+        If ignore_case is True, it indicates that ``first`` and ``second`` should be
+        compared case-insensitively.  See `Boolean  arguments` section for more details.
+        (This option is new in Robot Framework 3.0.1)
+
         If both arguments are multiline strings, the comparison is done using
         `multiline string comparisons`.
         """
         self._log_types_at_info_if_different(first, second)
+        if is_truthy(ignore_case):
+            if is_string(first) and is_string(second):
+                first = first.lower()
+                second = second.lower()
+            elif is_list_like(first) and is_list_like(second):
+                first = [x.lower() if type(x) is str else x for x in first]
+                second = [x.lower() if type(x) is str else x for x in second]
```

Comment:
I guess I was just trying to get to fancy here.  Thanks for pulling me back to reality.  
I've updated the doc string and removed the code per your suggestion.